### DISCUSSION TITLE
[9 gennaio 2023] Calcolabilità

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#2] [Calcolabilità]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/c353e96f-029d-4e55-869c-d82f66cb19af)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo la soluzione presente sui miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/c96888fe-b16a-4780-90bf-8c2c9a03ec9c)
----
La dimostrazione richiesta nel secondo punto è presente sul libro di testo

**Comment by aflaag** [comment]
Dimostrazione sulla falsa riga di quella di @Exyss 

Si prenda un $L_i \in \textsf{REC}$, e viene dunque riconosciuto da qualche $R_i$ riconoscitore tale che $L(R_i) = L_i$; per costruirne un decisore, è sufficiente considerare la seguente macchina $D_i$ che, su input $w \in \Sigma^*$, computi come segue:

- $M$ simula concorrentemente ogni macchina $R_j$ per ogni $j \in [1, k]$;
- poiché $L_1, \ldots, L_k$ partizionano $\Sigma^*$, una ed una sola macchina accetterà, e dunque:
  - se la macchina che accetta è proprio $R_i$, $D_i$ accetta
  - se la macchina che accetta è un'altra $R_j$ con $j \neq i$, $D_i$ rifiuta

Allora, per costruzione $R_i$ decide $L_i$; infine, poiché il ragionamento è applicabile ad ogni linguaggio, segue la tesi.

**Comment by daromacarlo** [comment]
<img width="1619" height="789" alt="Immagine 2026-02-02 173628" src="https://github.com/user-attachments/assets/a9e78b3e-a53f-4041-aba8-5ed562873935" />
<img width="1618" height="805" alt="Immagine 2026-02-02 173652" src="https://github.com/user-attachments/assets/598c13c7-733d-4c22-8718-c12211b7d5eb" />
<img width="1620" height="385" alt="Immagine 2026-02-02 173714" src="https://github.com/user-attachments/assets/4f6b39cc-ace7-43fb-bf23-7e03f57212c0" />
<img width="1621" height="304" alt="Immagine 2026-02-02 173726" src="https://github.com/user-attachments/assets/ca242c27-4a71-492b-bfdf-323f36218c40" />

**Reply by daromacarlo** [reply]
.
